### PR TITLE
Fix to use standard load balancer for azure

### PR DESCRIPTION
**Pull request by tei-k**
# Description
https://scalar-labs.atlassian.net/browse/DLT-8030

`Microsoft recommends Standard load balancer`
https://docs.microsoft.com/en-us/azure/load-balancer/skus

# Done
Fix to use `Standard Load Balancer` in Azure.

### CODE REVIEW

**Review by ymorimo — APPROVED** [review]
LGTM!

File: modules/azure/scalardl/envoy.tf [hunk 1]
```diff
@@ -88,7 +88,7 @@ resource "azurerm_public_ip" "envoy_nat_ip" {
 }
 
 resource "azurerm_nat_gateway" "envoy_natgw" {
-  count = local.envoy.enable_nlb && local.envoy.nlb_internal && length(local.locations) > 0 ? 1 : 0
+  count = local.envoy.enable_nlb && local.envoy.nlb_internal ? 1 : 0
```
Comment (feeblefakie) [inline modules/azure/scalardl/envoy.tf:91]: Just a question.
Does this change make envoy nodes use NAT for going out if both are true?

**Review by feeblefakie — APPROVED** [review]
LGTM!

File: modules/azure/scalardl/envoy.tf (hunk 1 above)
Comment (tei-k) [reply modules/azure/scalardl/envoy.tf:91]: Yes, If the `LB` is `Internal Standard SKU Load Balancer` and VM does not have  a public IP, use NAT for outbound connections.

**Pull request merged**